### PR TITLE
refactor(route53): favor `grant` method over precreated role

### PR DESCRIPTION
**Pull request by rix0rrr**
`PublicHostedZone` allows creating exactly one role for cross-account delegation.

Because of limitations of trust policy size, some teams inside Amazon are creating a role per opt-in region, in which case the "one precreated role" feature doesn't help at all.

Instead, add a `grantDelegation()` helper which works for both cases.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by ryparker — COMMENTED** [review]
Should we have a test for `.grantDelegation()`?

**Review by comcalvi — CHANGES_REQUESTED** [review]
:+1: to @rypaker's comment, would like to see a `grantDelegation()` test.

**Review by comcalvi — APPROVED** [review]

File: packages/@aws-cdk/aws-route53/lib/hosted-zone.ts [hunk 1]
```diff
@@ -342,6 +344,26 @@ export class PublicHostedZone extends HostedZone implements IPublicHostedZone {
       ttl: opts.ttl,
     });
   }
+
+  /**
+   * Grant permissions to add delegation records to this zone
+   */
+  public grantDelegation(grantee: iam.IGrantable) {
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-route53/lib/hosted-zone.ts:351]: from the build logs:

```
@aws-cdk/aws-msk: Build times for @aws-cdk/aws-msk: Total time (21.3s) | /codebuild/output/src805351294/src/github.com/aws/aws-cdk/node_modules/jsii/bin/jsii (9.6s) | cfn2ts (0.3s)
@aws-cdk/aws-route53: error: [awslint:grant-result:@aws-cdk/aws-route53.PublicHostedZone.grantDelegation] "grant" method must return @aws-cdk/aws-iam.Grant  (expected="@aws-cdk/aws-iam.Grant",actual="void")
```

Comment (comcalvi) [reply packages/@aws-cdk/aws-route53/lib/hosted-zone.ts:351]: Maybe worth adding an explicit return type? Up to you.

**Pull request merged**